File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        char[] chArr = code.toCharArray();
        boolean mode = false;
        String ret = "";
//        StringBuffer ret = new StringBuffer("");

        for (int i = 0; i < chArr.length; i++) {
            if (mode)
            {
                if (chArr[i] != '1' && i % 2 == 1)
                    ret += chArr[i];// .append(chArr[i]);
                else if (chArr[i] == '1')
                    mode = false;
            }
            else
            {
                if (chArr[i] != '1' && i % 2 == 0)
                    ret += chArr[i];//.append(chArr[i]);
                else if (chArr[i] == '1')
                    mode = true;
            }
        }
        return ret.equals("") ? "EMPTY" : ret;//.toString();
    }
}

File: 프로그래머스/Lv.0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    mode = 0;
    ret = "";
    for (let i = 0; i < code.length; i++)
    {
        if (mode === 0)
        {
            if (code[i] === "1")
                mode = 1;
            else if (i % 2 === 0) 
                ret += code[i];
        }
        else {
            if (code[i] === "1")
                mode = 0;
            else if (i % 2 === 1)
                ret += code[i];
        }
    }
    return ret.length === 0 ? "EMPTY" : ret;
}